internal/machine/query.go

```diff
@@ -22,3 +22,15 @@ func ListActive(ctx context.Context) ([]*api.Machine, error) {
 
 	return machines, nil
 }
+
+// AppV2ListActive lists machines that are part of v2 apps platform
+func AppV2ListActive(ctx context.Context) ([]*api.Machine, error) {
+	machines, err := ListActive(ctx)
+	if err != nil {
+		return nil, err
+	}
+	machines = lo.Filter(machines, func(m *api.Machine, _ int) bool {
+		return m.IsFlyAppsPlatform()
+	})
+	return machines, nil
+}
```

Comment:
Does this do the same thing as `flaps.ListFlyAppsMachines`?

Reply:
Yes, kind of. I think flaps and `internal/machine/query.go` needs so factoring to keep concerns separate but you are right that is basically doing the same.

there is also https://github.com/superfly/flyctl/blob/68223a9c393f91de16eddb74fc7be70890c99cef/internal/machine/query.go#L11

I didn't want to change either of them until having a better picture of how these functions are used, so added a new function, sorry.

Reply:
addressed in 005d53726d0ad71594d1a4f2d60cba6a1de1e13a